### PR TITLE
#BAH-3019 | Add. Backup and Restore Scripts for Bahmni databases and files

### PR DESCRIPTION
**Pull request by deeptirawat1510**
Co-authored by: Riya Kumari, Deepti Rawat

### CODE REVIEW

**Comment by gsluthra** [comment]
Looks good from my side. Will wait for @mohan-13 or @SooryaKumaranC-tw to provide comments. This ticket takes care of some of the containers (EMR). Might need a separate JIRA ticket for ELIS, Odoo, Dcm4chee, etc. But this is a good start.

File: backups/restore.sh [hunk 1]
```diff
@@ -0,0 +1,137 @@
+#!/bin/bash
+
+# Function to restore OpenMRS database
+function restore_openmrs_database() {
+  if [[ -d "$restore_folder/openmrs" ]]; then
+    echo "Restoring OpenMRS database"
+    docker exec -i "$openmrs_container" mysql -u $openmrsdb_username --password=$openmrsdb_password openmrs < "$openmrs_backup_file"
```
Comment (SooryaKumaranC-tw) [inline backups/restore.sh:7]: Hope you have tested for restoring the stored procedures

Comment (deeptirawat1510) [reply backups/restore.sh:7]: Yes @SooryaKumaranC-tw we tested, able to restore the stored procedure.

Comment (SooryaKumaranC-tw) [reply backups/restore.sh:7]: Okay. Thanks

File: backups/restore.sh [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+  else
+    echo "Openmrs backup folder is missing. Thus, skipping OpenMRS restoration."
+  fi
+}
+
+# Function to restore Reports database
+function restore_reports_database() {
+  if [[ -d "$restore_folder/reports" ]]; then
+    echo "Restoring Reports database"
+    docker exec -i "$reports_container" mysql -u $reportsdb_username --password=$reportsdb_password bahmni_reports < "$reports_backup_file"
```
Comment (SooryaKumaranC-tw) [inline backups/restore.sh:17]: Should we check for the file "reports_backup_file" too and display message accordingly.

Comment (deeptirawat1510) [reply backups/restore.sh:17]: Considering folders will only be created if backup is taken so we only checked for folders and not the file.

Comment (SooryaKumaranC-tw) [reply backups/restore.sh:17]: Okay thanks.

**Review by SooryaKumaranC-tw — APPROVED** [review]

**Comment by gsluthra** [comment]
Is this good to merge now? @deeptirawat1510 @mohan-13 @SooryaKumaranC-tw

**Review by mohan-13 — APPROVED** [review]

**Pull request merged**